[BUG] test_parquet_pyarrow_data_page_version timeout intermittently

**Describe the bug**
first seen in rapids_databricks_nightly-dev-github run: 1633

```
Databricks / Matrix - DB_RUNTIME = '13.3' / Build / src.main.python.parquet_pyarrow_test.test_parquet_pyarrow_data_page_version[2.0][DATAGEN_SEED=1745902584, TZ=UTC, IGNORE_ORDER({'local': True})]
Databricks / Matrix - DB_RUNTIME = '13.3' / Build / src.main.python.parquet_pyarrow_test.test_parquet_pyarrow_data_page_version[2.0][DATAGEN_SEED=1745902584, TZ=UTC, IGNORE_ORDER({'local': True})]
Databricks / Matrix - DB_RUNTIME = '14.3' / Build / src.main.python.parquet_pyarrow_test.test_parquet_pyarrow_data_page_version[2.0][DATAGEN_SEED=1745902652, TZ=UTC, IGNORE_ORDER({'local': True})]
Databricks / Matrix - DB_RUNTIME = '14.3' / Build / src.main.python.parquet_pyarrow_test.test_parquet_pyarrow_data_page_version[2.0][DATAGEN_SEED=1745902652, TZ=UTC, IGNORE_ORDER({'local': True})]
```

```
spark_tmp_path = '/tmp/pyspark_tests//0429-043259-1u94v4ov-10-0-128-17-gw7-72146-2045617302/'
data_page_version = '2.0'

    @pyarrow_test
    @ignore_order(local=True)
    @pytest.mark.parametrize('data_page_version', ['1.0', '2.0'])
    def test_parquet_pyarrow_data_page_version(
            spark_tmp_path,
            data_page_version):
        gen_list = [('_c' + str(i), gen) for i, gen in enumerate(sub_gens)]
        pyarrow_write_conf = {'data_page_version': data_page_version}
>       assert_gpu_and_pyarrow_are_compatible(spark_tmp_path, gen_list, conf=_common_rebase_conf, pyarrow_write_conf=pyarrow_write_conf)

../../src/main/python/parquet_pyarrow_test.py:259: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/main/python/parquet_pyarrow_test.py:85: in assert_gpu_and_pyarrow_are_compatible
    assert_gpu_and_cpu_are_equal_collect(read_parquet(pyarrow_parquet_path), conf=conf)
../../src/main/python/asserts.py:599: in assert_gpu_and_cpu_are_equal_collect
    _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first, result_canonicalize_func_before_compare=result_canonicalize_func_before_compare)
../../src/main/python/asserts.py:507: in _assert_gpu_and_cpu_are_equal
    from_gpu = run_on_gpu()
../../src/main/python/asserts.py:500: in run_on_gpu
    from_gpu = with_gpu_session(bring_back, conf=conf)
../../src/main/python/spark_session.py:167: in with_gpu_session
    return with_spark_session(func, conf=copy)
/usr/lib/python3.10/contextlib.py:79: in inner
    return func(*args, **kwds)
../../src/main/python/spark_session.py:134: in with_spark_session
    ret = func(_spark)
../../src/main/python/asserts.py:209: in <lambda>
    bring_back = lambda spark: limit_func(spark).collect()
/databricks/spark/python/pyspark/instrumentation_utils.py:48: in wrapper
    res = func(*args, **kwargs)
/databricks/spark/python/pyspark/sql/dataframe.py:1244: in collect
    sock_info = self._jdf.collectToPython()
/databricks/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py:1355: in __call__
    return_value = get_return_value(
/databricks/spark/python/pyspark/errors/exceptions/captured.py:188: in deco
    return f(*a, **kw)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

answer = 'xro7248'
gateway_client = <py4j.clientserver.JavaClient object at 0x7fd8b62ca770>
target_id = 'o7245', name = 'collectToPython'

    def get_return_value(answer, gateway_client, target_id=None, name=None):
        """Converts an answer received from the Java gateway into a Python object.
    
        For example, string representation of integers are converted to Python
        integer, string representation of objects are converted to JavaObject
        instances, etc.
    
        :param answer: the string returned by the Java gateway
        :param gateway_client: the gateway client used to communicate with the Java
            Gateway. Only necessary if the answer is a reference (e.g., object,
            list, map)
        :param target_id: the name of the object from which the answer comes from
            (e.g., *object1* in `object1.hello()`). Optional.
        :param name: the name of the member from which the answer comes from
            (e.g., *hello* in `object1.hello()`). Optional.
        """
        if is_error(answer)[0]:
            if len(answer) > 1:
                type = answer[1]
                value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
                if answer[1] == REFERENCE_TYPE:
>                   raise Py4JJavaError(
                        "An error occurred while calling {0}{1}{2}.\n".
                        format(target_id, ".", name), value)
E                   py4j.protocol.Py4JJavaError: An error occurred while calling o7245.collectToPython.
E                   : org.apache.spark.SparkException: [SPARK_JOB_CANCELLED] Job 133 cancelled RAPIDS Integration Test Job 133 exceeded the timeout of 3600 seconds, cancelling. Look into fixing the test or reducing its execution time. If necessary, adjust the timeout using the marker pytest.mark.spark_job_timeout(seconds,dump_threads)
E                   	at org.apache.spark.scheduler.DAGScheduler.handleJobCancellation(DAGScheduler.scala:3541)
E                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3892)
E                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:3851)
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/requirements.txt
+++ b/integration_tests/requirements.txt
@@ -14,7 +14,7 @@
 pytest
 sre_yield
 pandas
-pyarrow == 17.0.0 ; python_version >= '3.8'
+pyarrow == 17.0.0 ; python_version == '3.8'
 pyarrow == 19.0.1 ; python_version >= '3.9'
 pytest-xdist >= 2.0.0
 findspark

--- a/integration_tests/requirements.txt
+++ b/integration_tests/requirements.txt
@@ -14,7 +14,7 @@
 pytest
 sre_yield
 pandas
-pyarrow
+pyarrow == 19.0.1
 pytest-xdist >= 2.0.0
 findspark
 fsspec == 2025.3.0

--- a/integration_tests/requirements.txt
+++ b/integration_tests/requirements.txt
@@ -14,7 +14,8 @@
 pytest
 sre_yield
 pandas
-pyarrow == 19.0.1
+pyarrow == 17.0.0 ; python_version >= '3.8'
+pyarrow == 19.0.1 ; python_version >= '3.9'
 pytest-xdist >= 2.0.0
 findspark
 fsspec == 2025.3.0

--- a/jenkins/databricks/setup.sh
+++ b/jenkins/databricks/setup.sh
@@ -24,6 +24,8 @@ sudo apt-get install -y zip
 # Configure spark environment on Databricks
 export SPARK_HOME=$DB_HOME/spark
 
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
 # Workaround to support local spark job
 sudo ln -sf $DB_HOME/jars/ $SPARK_HOME/jars
 
@@ -54,8 +56,6 @@ fi
 # Set the path of python site-packages, and install packages here.
 PYTHON_SITE_PACKAGES="$HOME/.local/lib/python${PYTHON_VERSION}/site-packages"
 # Use "python -m pip install" to make sure pip matches with python.
-$PYSPARK_PYTHON -m pip install --target $PYTHON_SITE_PACKAGES pytest sre_yield requests pandas pyarrow findspark pytest-xdist pytest-order
+$PYSPARK_PYTHON -m pip install --target $PYTHON_SITE_PACKAGES -r ${SCRIPTPATH}/../../integration_tests/requirements.txt
+$PYSPARK_PYTHON -m pip install --target $PYTHON_SITE_PACKAGES requests pytest-order
 
-# Install fastparquet (and numpy as its dependency).
-echo -e "fsspec==2025.3.0\nfastparquet==0.8.3;python_version=='3.8'\nfastparquet==2024.5.0;python_version>='3.9'" > fastparquet.txt
-$PYSPARK_PYTHON -m pip install --target $PYTHON_SITE_PACKAGES -r fastparquet.txt

--- a/jenkins/databricks/setup.sh
+++ b/jenkins/databricks/setup.sh
@@ -24,7 +24,7 @@ sudo apt-get install -y zip
 # Configure spark environment on Databricks
 export SPARK_HOME=$DB_HOME/spark
 
-SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+SCRIPTPATH="$(cd -P "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
 # Workaround to support local spark job
 sudo ln -sf $DB_HOME/jars/ $SPARK_HOME/jars

